bom/pom.xml

```diff
@@ -24,7 +24,7 @@
         <!-- compile/runtime dependency versions, managed in the BOM -->
         <classmateVersion>1.0.0</classmateVersion>
         <ehcacheVersion>2.6.9</ehcacheVersion>
-        <infinispanVersion>8.0.2.Final</infinispanVersion>
+        <infinispanVersion>8.2.0.Final</infinispanVersion>
```

Comment:
N.B. my goal of "align with WildFly 10" applies to the JTA API only.

Narayana 5.2.12 is indeed what was in WF 10.0.0.Final, but the 5.2.14.Final is now going to replace it.. micro bugfix release so it shouldbn't matter for us.
I can make it 5.2.12 if you prefer ..

Regarding Infinispan: in this case we do not follow WildFly versions but track the Infinispan community, so that upgrade is a different JIRA; I included it in this PR as it requires the JTA upgrade.